src/pages/LoginPage.js

```diff
@@ -1,9 +1,93 @@
-import React from 'react'
+import React from "react";
+import { ThemeButton } from "../components/ThemeButton";
+import { useForm } from "react-hook-form";
+import {
+  TextField,
+  Container,
+  Grid,
+  Link,
+  CircularProgress,
+  Button,
+} from "@material-ui/core";
+import { makeStyles } from "@material-ui/core/styles";
+import { Link as RouterLink } from "react-router-dom";
+import { login } from "../mock/auth";
 
 export const LoginPage = () => {
+  const {
+    register,
+    handleSubmit,
+    errors,
+    formState: { isSubmitting },
+  } = useForm();
+  const onSubmit = async (data) => {
+    console.log(data);
+    login();
+    const response = await new Promise((resolve) => {
+      setTimeout(() => resolve(true), 10000);
+    });
+  };
+  const classes = useStyles();
+
   return (
-    <div>
-      LoginPage
-    </div>
-  )
-}
\ No newline at end of file
+    <Container className={classes.container} maxWidth="xs">
+      <form onSubmit={handleSubmit(onSubmit)}>
+        <Grid
+          spacing={2}
+          container
+          direction="column"
+          justify="space-between"
+          alignItems="center"
+        >
+          <Grid item>
+            <TextField
+              label="Username"
+              name="username"
+              inputRef={register({ required: "Du må gi et brukernavn!" })}
+              error={!!errors.username}
+              helperText={errors.username ? errors.username.message : ""}
+            />
+          </Grid>
+          <Grid item>
+            <TextField
+              label="Password"
+              name="password"
+              inputRef={register({ required: "Du må gi et passord!" })}
+              error={!!errors.password}
+              helperText={errors.password ? errors.password.message : ""}
+              type="password"
+            />
+          </Grid>
+          <Grid item>
+            <ThemeButton
+              text={isSubmitting ? <CircularProgress size={20} /> : "Log in"}
```

Comment:
For å se CircularProgress bedre går det an å sette `color="secondary"`

Reply:
Ja, det var bedre :)